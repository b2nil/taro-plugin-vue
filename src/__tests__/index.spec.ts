import {
  isMiniappNativeTag,
  transformMiniappAssetUrls,
  transformH5AssetUrls,
  transformH5Tags,
  transformEnv,
  genVueOptions,
  vuePlugin
} from '..'

describe('genVueOptions', () => {
  test('should generate default mini-app options', () => {
    const options = genVueOptions()

    expect(
      options.template!.compilerOptions!.isNativeTag
    ).toEqual(isMiniappNativeTag)

    expect(
      options.template!.transformAssetUrls
    ).toEqual(transformMiniappAssetUrls)

    expect(options).toMatchSnapshot()
  })

  test('should merge user passed mini-app options', () => {
    const options = genVueOptions({
      template: {
        compilerOptions: {
          nodeTransforms: [transformEnv()]
        }
      }
    })

    expect(
      options.template!.compilerOptions!.isNativeTag
    ).toEqual(isMiniappNativeTag)

    expect(
      options.template!.compilerOptions!.nodeTransforms![0].toString()
    ).toEqual(transformEnv().toString())

    expect(
      options.template!.transformAssetUrls
    ).toEqual(transformMiniappAssetUrls)

    expect(options).toMatchSnapshot()
  })

  test('should generate default h5 options', () => {
    const options = genVueOptions({ h5: true })
    expect(
      options.template!.compilerOptions!.nodeTransforms![0].toString()
    ).toEqual(transformH5Tags().toString())

    expect(
      options.template!.transformAssetUrls
    ).toEqual(transformH5AssetUrls)

    expect(options).toMatchSnapshot()
  })

  test('should merge user passed h5 options', () => {
    const options = genVueOptions({
      h5: true,
      template: {
        compilerOptions: {
          nodeTransforms: [transformEnv('h5')]
        }
      }
    })

    expect(
      options.template!.compilerOptions!.nodeTransforms![0].toString()
    ).toEqual(transformH5Tags().toString())

    expect(
      options.template!.compilerOptions!.nodeTransforms![1].toString()
    ).toEqual(transformEnv().toString())

    expect(
      options.template!.transformAssetUrls
    ).toEqual(transformH5AssetUrls)

    expect(options).toMatchSnapshot()
  })
})

test('vuePlugin', () => {
  const plugin = vuePlugin()
  expect(plugin).toMatchSnapshot()
})