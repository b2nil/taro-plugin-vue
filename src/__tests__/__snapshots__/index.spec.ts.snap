// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genVueOptions should generate default h5 options 1`] = `
Object {
  "h5": true,
  "template": Object {
    "compilerOptions": Object {
      "mode": "module",
      "nodeTransforms": Array [
        [Function],
      ],
      "optimizeImports": true,
    },
    "transformAssetUrls": Object {
      "audio": Array [
        "src",
      ],
      "cover-image": Array [
        "src",
      ],
      "image": Array [
        "src",
      ],
      "live-player": Array [
        "src",
      ],
      "source": Array [
        "src",
      ],
      "taro-audio": Array [
        "src",
      ],
      "taro-cover-image": Array [
        "src",
      ],
      "taro-image": Array [
        "src",
      ],
      "taro-live-player": Array [
        "src",
      ],
      "taro-source": Array [
        "src",
      ],
      "taro-video": Array [
        "src",
        "poster",
      ],
      "video": Array [
        "src",
        "poster",
      ],
    },
  },
}
`;

exports[`genVueOptions should generate default mini-app options 1`] = `
Object {
  "template": Object {
    "compilerOptions": Object {
      "isNativeTag": [Function],
      "mode": "module",
      "optimizeImports": true,
    },
    "transformAssetUrls": Object {
      "audio": Array [
        "src",
      ],
      "cover-image": Array [
        "src",
      ],
      "image": Array [
        "src",
      ],
      "live-player": Array [
        "src",
      ],
      "source": Array [
        "src",
      ],
      "video": Array [
        "src",
        "poster",
      ],
    },
  },
}
`;

exports[`genVueOptions should merge user passed h5 options 1`] = `
Object {
  "h5": true,
  "template": Object {
    "compilerOptions": Object {
      "mode": "module",
      "nodeTransforms": Array [
        [Function],
        [Function],
      ],
      "optimizeImports": true,
    },
    "transformAssetUrls": Object {
      "audio": Array [
        "src",
      ],
      "cover-image": Array [
        "src",
      ],
      "image": Array [
        "src",
      ],
      "live-player": Array [
        "src",
      ],
      "source": Array [
        "src",
      ],
      "taro-audio": Array [
        "src",
      ],
      "taro-cover-image": Array [
        "src",
      ],
      "taro-image": Array [
        "src",
      ],
      "taro-live-player": Array [
        "src",
      ],
      "taro-source": Array [
        "src",
      ],
      "taro-video": Array [
        "src",
        "poster",
      ],
      "video": Array [
        "src",
        "poster",
      ],
    },
  },
}
`;

exports[`genVueOptions should merge user passed mini-app options 1`] = `
Object {
  "template": Object {
    "compilerOptions": Object {
      "isNativeTag": [Function],
      "mode": "module",
      "nodeTransforms": Array [
        [Function],
      ],
      "optimizeImports": true,
    },
    "transformAssetUrls": Object {
      "audio": Array [
        "src",
      ],
      "cover-image": Array [
        "src",
      ],
      "image": Array [
        "src",
      ],
      "live-player": Array [
        "src",
      ],
      "source": Array [
        "src",
      ],
      "video": Array [
        "src",
        "poster",
      ],
    },
  },
}
`;

exports[`vuePlugin 1`] = `
Object {
  "config": [Function],
  "configResolved": [Function],
  "configureServer": [Function],
  "handleHotUpdate": [Function],
  "load": [Function],
  "name": "vite:vue",
  "resolveId": [Function],
  "transform": [Function],
}
`;
